### PR TITLE
Fix Lead that's missing enumerator disposal & test

### PR DESCRIPTION
**Pull request by Orace**
Fix a part of #649
Fork of #660

### CODE REVIEW

File: MoreLinq.Test/LeadTest.cs [hunk 1]
```diff
@@ -1,6 +1,25 @@
+#region License and Terms
```
Comment (atifaziz) [inline MoreLinq.Test/LeadTest.cs]: Thanks for adding this header but it belongs in a separate PR.

**Comment by Orace** [comment]
Closed. The implementation of Lead is fixed here #672
